inc/networkdiscovery.class.php

```diff
@@ -327,18 +327,31 @@ function run($jobstate) {
       $pfTaskjoblog = new PluginFusioninventoryTaskjoblog();
       $pfIPRange = new PluginFusioninventoryIPRange();
       $pfToolbox = new PluginFusioninventoryToolbox();
+      $pfConfig = new PluginFusioninventoryConfig();
 
       $pfAgent->getFromDB($jobstate->fields['plugin_fusioninventory_agents_id']);
 
       $sxml_option = $this->message->addChild('OPTION');
       $sxml_option->addChild('NAME', 'NETDISCOVERY');
 
       $sxml_param = $sxml_option->addChild('PARAM');
-      $sxml_param->addAttribute('THREADS_DISCOVERY',
-         $pfAgent->fields["threads_networkdiscovery"]);
-      $sxml_param->addAttribute('TIMEOUT',
-         $pfAgent->fields["timeout_networkdiscovery"]);
-      $sxml_param->addAttribute('PID', $jobstate->fields['id']);
+      // Use general config when threads number is set to 0 on the agent
+      if ($pfAgent->fields["threads_networkdiscovery"]<1) {
```

Comment:
Better to use `== 0` instead `< 1`

Reply:
Okay, that should be clearer. I changed the 4 case where I used `<1` check.